### PR TITLE
MGMT-9304: continuous upload of bootstrap logs

### PR DESCRIPTION
**Pull request by slaviered**
Some cases in triage are missing the bootstrap logs.

While in some cases, we can prove that an automated
system shut down the node, thus preventing the agent
from uploading the logs in other cases, we can not
assert why logs are missing.

Continuous uploading of logs will help us at least get
some of the data

### CODE REVIEW

File: src/logs_sender/send_logs.go [hunk 1]
```diff
@@ -268,25 +305,24 @@ func getJournalLogs(l LogsSender, since string, outputFilePath string, journalFi
...
-	log.Infof("Archiving %s and creating %s", inputPath, outputFile)
-	args := []string{"-czvf", outputFile, "-C", filepath.Dir(inputPath), filepath.Base(inputPath)}
+func uploadLogs(l LogsSender, inputPath string, archivePath string) error {
+	if lerr := l.LogProgressReport(models.LogsStateCollecting); lerr != nil {
+		log.WithError(lerr).Error("failed to send log progress collecting to service")
+	}
 
-	_, err, execCode := l.Execute("tar", args...)
+	log.Infof("Archiving %s and creating %s", inputPath, archivePath)
+	args := []string{"-czvf", archivePath, "-C", filepath.Dir(inputPath), filepath.Base(inputPath)}
```
[the remark this answers is not in the document]
Comment (omertuc) [reply src/logs_sender/send_logs.go:314]: (the first happens in `CollectPartialLogs`)

File: src/logs_sender/send_logs.go [hunk 2]
```diff
@@ -127,11 +126,49 @@ func (e *LogsSenderExecuter) LogProgressReport(infraEnvID strfmt.UUID, hostID st
...
+		case <-ticker.C:
+			//collect partial logs from installer gather
+			args := []string{"-czvf", partialArchivePath, "-C", filepath.Dir(inputPath), filepath.Base(inputPath)}
+			_, errOut, execCode := e.Execute("tar", args...)
+			if execCode != 0 {
+				log.WithError(errors.Errorf(errOut)).Errorf("Failed to run to archive %s.", inputPath)
+				continue
+			}
+			log.Info("uploading partial logs...")
+			err := uploadLogs(e, targetDir, fmt.Sprintf("%s/logs.tar.gz", logsDir))
```
Comment (omertuc) [inline src/logs_sender/send_logs.go:154]: Shouldn't this be `partialArchivePath` ?

File: src/logs_sender/send_logs.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (slaviered) [reply src/logs_sender/send_logs.go:314]: No. In the partial log collection we are archiving the temporary directory of install-gather.sh into a tar and then put it in the inputPath. Here we are archiving the entire inputPath directory (a path that includes our own logs, partial intall-gather or the full install-gather tar and sending the result to the service

**Pull request merged**